src/openslide-vendor-philips-isyntax.c

```diff
@@ -0,0 +1,397 @@
+        return true;
+    }
+
+    LOG("not isyntax.");
+    return false;
+}
+
+static bool philips_isyntax_read_tile(
+        openslide_t *osr,
+        cairo_t *cr,
+        struct _openslide_level *osr_level,
+        int64_t tile_col, int64_t tile_row,
+        void *arg G_GNUC_UNUSED,
+        GError **err G_GNUC_UNUSED) {
+    philips_isyntax_t *data = osr->data;
+    isyntax_t* isyntax = data->isyntax;
+
+    philips_isyntax_level* pi_level = (philips_isyntax_level*)osr_level;
+
+    // LOG("level=%d tile_col=%ld tile_row=%ld", pi_level->level_idx, tile_col, tile_row);
+    // tile size
+    int64_t tw = libisyntax_get_tile_width(isyntax);
+    int64_t th = libisyntax_get_tile_height(isyntax);
+
+    // Openslide cache
+    g_autoptr(_openslide_cache_entry) cache_entry = NULL;
+    uint32_t *tiledata = _openslide_cache_get(osr->cache,
+                                              pi_level, tile_col, tile_row,
+                                              &cache_entry);
+    if (!tiledata) {
+        int scale = libisyntax_level_get_scale(pi_level->isyntax_level);
+        ASSERT_OK(libisyntax_tile_read(isyntax, data->cache->cache, scale, tile_col, tile_row, &tiledata));
+        annotate_tile(tiledata, scale, tile_col, tile_row, tw, th);
+
+        _openslide_cache_put(osr->cache, pi_level, tile_col, tile_row,
+                             tiledata, tw * th * 4,
+                             &cache_entry);
+    }
+
+    // draw it
+    g_autoptr(cairo_surface_t) surface =
+            cairo_image_surface_create_for_data((unsigned char *) tiledata,
+                                                CAIRO_FORMAT_ARGB32,
+                                                tw, th, tw * 4);
+    cairo_set_source_surface(cr, surface, 0, 0);
+    // https://lists.cairographics.org/archives/cairo/2012-June/023206.html
+    // Those are the operators that are observed to work:
+    //    w CAIRO_OPERATOR_SATURATE (current_cairo_operator, aka default OpenSlide)
+    //    w CAIRO_OPERATOR_OVER ("This operator is cairo's default operator."),
+    //    w CAIRO_OPERATOR_DEST_OVER,
+    // SATURATE takes ~12sec to read a dummy slide (forcing all tiles to not exist), OVER & DEST_OVER take 3.5 sec
+    // for same setup. Selecting OVER as the Cairo's default operator, the three outputs are identical.
+    cairo_operator_t current_cairo_operator = cairo_get_operator(cr);
+    cairo_set_operator(cr, CAIRO_OPERATOR_OVER);
+    cairo_paint(cr);
+    cairo_set_operator(cr, current_cairo_operator);
+    return true;
+}
+
+static void add_float_property(openslide_t *osr, const char* property_name, float value) {
+    g_hash_table_insert(osr->properties, g_strdup(property_name),
+                        _openslide_format_double(value));
+}
+
+static bool philips_isyntax_open(
+        openslide_t *osr,
+        const char *filename,
+        struct _openslide_tifflike *tl G_GNUC_UNUSED,
+        struct _openslide_hash *quickhash1 G_GNUC_UNUSED,
```

Comment:
Done, with `_openslide_hash_disable()`.

This overlaps somewhat with the properties/metadata discussion below. Do you know what this hash is being used in practice? I was thinking just hashing the filename (no path), or filename + dimensions. But tbh I'd rather leave it disabled if it is unclear what sense of uniqueness/equality we must provide.

Reply:
I know some applications have used it in the past, but I don't know how widely used it is.  The [design goal](https://openslide.org/docs/newformat/#quickhash-1) is to uniquely identify a particular slide, e.g. to help applications deduplicate multiple uploads of the same slide.

The filename should not be part of the quickhash, since the user might rename a slide file.  We often hash the slide metadata plus the pixels from the lowest-resolution pyramid level.  If the latter is expensive in iSyntax, and the metadata is expressive enough, just the metadata may be sufficient.